A/reduce mass attempts (#37)
ministryofjustice/opg-modernising-lpa-docs: src/research-development/articles/reducing-mass-attempts.md

---
layout: layouts/article.njk
title: "Using Technology to reduce mass LPA attempts"
date: 2022-02-21
tags: research_development
areas_of_coverage: ["Digital Service"]
contributors: ["John Nolan", "Andrew Pearce"]
---

## Introduction

OPG needs to cover costs. ID lookups will cost money, as do other elements of maintaining the service.

Registered and unused LPAs cover costs of a small number of expensive investigations/safeguard work.

Regardless of the solution we come to, we need to identify ways in which we can mitigate abuse of bad actors abusing a service and costing the business, and in turn our end users, money.

We should also try to protect the highest cost endpoints ensuring as much of the user journey is unaffected by any negative experiences.

## Solutions

Below is a list of options we can investigate to try and mitigate this problem.

### Account Creation Flow

> There are ways to prevent creation of accounts and use of verification flows on the service.

Reviewer: ```suggestion
There are ways to prevent malicious creation of accounts and use of verification flows on the service.
```

By setting up our registration flow in the correct way, we can put manual steps in place to prevent automation attempts without causing a bad experience to the end user.

Some of these are

- Limit the number of LPAs allowed to be created over time. A rate limit to prevent bad behaviour

> - One account per email address

Reviewer: ```suggestion
- Permit only one account per email address
```

> - Email alias prevention or instances of alias check

Reviewer: ```suggestion
- Check for or prevent the use of email aliasing
```
